### ISSUE TITLE
The "stable" monodevelop version crashes very frequently.

### ISSUE THREAD

**Issue by jkpubsrc** [mono/monodevelop#4118]
The "stable" monodevelop version crashes very frequently. After a few attempts of compilation it crashes. MonoDevelop is very unstable: This happens about 30% of all compilations on two of my systems here. Both are set up similar, running Ubuntu Mate, one the most recent, the other a 16.04.

I couldn't find a way to clearly reproduce this problem. The behavior seems to be indeterministic.

The error that seems to have caused the crash:

```
The application was terminated by a signal: SIGABRT
FATAL ERROR [2018-03-11 18:56:12Z]: An unhandled exception has occured. Terminating MonoDevelop? True
System.InvalidOperationException: An attempt was made to transition a task to a final state when it had already completed.
  at System.Threading.Tasks.TaskCompletionSource`1[TResult].SetException (System.Exception exception) [0x00021] in <f56c876907e742b0aa586f051fcce845>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.NotifyResponse (MonoDevelop.Core.Execution.RemoteProcessConnection+MessageRequest req, MonoDevelop.Core.Execution.BinaryMessage res) [0x000a0] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.AbortPendingMessages () [0x00043] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.AbortConnection (System.String message, System.Boolean isAsync) [0x00010] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.PostMessageInternal (MonoDevelop.Core.Execution.BinaryMessage message) [0x000bb] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.SendMessages () [0x00070] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.<PostMessage>b__65_0 (System.Object <p0>) [0x00001] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <f56c876907e742b0aa586f051fcce845>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: An attempt was made to transition a task to a final state when it had already completed.
  at System.Threading.Tasks.TaskCompletionSource`1[TResult].SetException (System.Exception exception) [0x00021] in <f56c876907e742b0aa586f051fcce845>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.NotifyResponse (MonoDevelop.Core.Execution.RemoteProcessConnection+MessageRequest req, MonoDevelop.Core.Execution.BinaryMessage res) [0x000a0] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.AbortPendingMessages () [0x00043] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.AbortConnection (System.String message, System.Boolean isAsync) [0x00010] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.PostMessageInternal (MonoDevelop.Core.Execution.BinaryMessage message) [0x000bb] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.SendMessages () [0x00070] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at MonoDevelop.Core.Execution.RemoteProcessConnection.<PostMessage>b__65_0 (System.Object <p0>) [0x00001] in <9b47a1f45a7d48b9a71e86b27c89ec26>:0
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <f56c876907e742b0aa586f051fcce845>:0
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <f56c876907e742b0aa586f051fcce845>:0
```

No offence ment, but according to the frequency of this problem I'm wondering if the development version of MonoDevelop is more stable than the actual stable version :-/

Do you require more information? If I can be of any more help please let me know.

> VS bug [#580790](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/580790)

**Comment by jkpubsrc** [comment]
I've the feeling it's GUI related. If I try to not do Alt+Tab to a different window during compilation the stability increases a bit, maybe to about only having 10% crashes of all built attempts.

**Comment by KaaCee** [comment]
I can confirm this issue.
I'm new to MonoDevelop, but it's really hard to use it productively because of the frequent crashes.

**Comment by jkpubsrc** [comment]
What's that VS bug s.o. added to my post? I've no Microsoft account and I don't intend to get one.

**Comment by slluis** [comment]
You can ignore that VS bug tag, it is just for our internal work item tracker.

**Comment by KaaCee** [comment]
Did anyone find a workaround?
This is really annoying.
Crash and restart almost every 5 minutes.

**Comment by jkpubsrc** [comment]
"Anoying" is an euphemism.

**Comment by slluis** [comment]
Fixed in version 7.5.1.9 (d15-7)

Author: Lluis Sanchez
Commit: de2c8c1fad3f8184f42b7521cea454a9096e3686 (mono/monodevelop)

**Comment by slluis** [comment]
The above comment was supposed to be added automatically, but we currently have some issues with the bug closing hook. I hope this helps recovering a bit your enthusiasm, sorry for trouble!

**Comment by KaaCee** [comment]
Hi slluis,

thanks for this information. I will test this asap.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-18] Comment by **Lluis Sanchez Gual**

Testing

**Comment by dankrusi** [comment]
I have been facing a similar issue for several months now (even on MonoDevelop 7.6 build 711):

=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

FATAL ERROR [2018-07-18 15:41:55Z]: An unhandled exception has occurred. Terminating MonoDevelop? True
System.InvalidOperationException
  at System.Threading.Tasks.TaskCompletionSource`1[TResult].SetException (System.Exception exception) <0x7f9229b02f50 + 0x00056> in <040e20dd5607491e942c086f20c1890d>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.NotifyResponse (MonoDevelop.Core.Execution.RemoteProcessConnection+MessageRequest req, MonoDevelop.Core.Execution.BinaryMessage res) [0x000a0] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.AbortPendingMessages () [0x00043] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.AbortConnection (System.String message, System.Boolean isAsync) [0x00010] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.PostMessageInternal (MonoDevelop.Core.Execution.BinaryMessage message) [0x000bb] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.SendMessages () [0x00070] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.<PostMessage>b__65_0 (System.Object <p0>) [0x00001] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) <0x7f9229afc170 + 0x0002a> in <040e20dd5607491e942c086f20c1890d>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) <0x7f9229af6f30 + 0x001a2> in <040e20dd5607491e942c086f20c1890d>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () <0x7f9229afc120 + 0x00025> in <040e20dd5607491e942c086f20c1890d>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <040e20dd5607491e942c086f20c1890d>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () <0x7f9229afbfa0 + 0x00005> in <040e20dd5607491e942c086f20c1890d>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException
  at System.Threading.Tasks.TaskCompletionSource`1[TResult].SetException (System.Exception exception) <0x7f9229b02f50 + 0x00056> in <040e20dd5607491e942c086f20c1890d>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.NotifyResponse (MonoDevelop.Core.Execution.RemoteProcessConnection+MessageRequest req, MonoDevelop.Core.Execution.BinaryMessage res) [0x000a0] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.AbortPendingMessages () [0x00043] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.AbortConnection (System.String message, System.Boolean isAsync) [0x00010] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.PostMessageInternal (MonoDevelop.Core.Execution.BinaryMessage message) [0x000bb] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.SendMessages () [0x00070] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at MonoDevelop.Core.Execution.RemoteProcessConnection.<PostMessage>b__65_0 (System.Object <p0>) [0x00001] in <eaa41b6ba717401bb0842aa80ee5d521>:0 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) <0x7f9229afc170 + 0x0002a> in <040e20dd5607491e942c086f20c1890d>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) <0x7f9229af6f30 + 0x001a2> in <040e20dd5607491e942c086f20c1890d>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () <0x7f9229afc120 + 0x00025> in <040e20dd5607491e942c086f20c1890d>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <040e20dd5607491e942c086f20c1890d>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () <0x7f9229afbfa0 + 0x00005> in <040e20dd5607491e942c086f20c1890d>:0 


The crash happens when building and running very consistently (about every 5th time).


=== MonoDevelop ===

Version 7.6 Preview (7.6 build 711)
Installation UUID: 2358209e-78ee-46b6-a868-a2533d332305
Runtime:
	Mono 5.14.0.167 (tarball Mon Jul 16 09:12:53 UTC 2018) (64-bit)
	GTK+ 2.24.32 (Mint-Y-Dark theme)

=== NuGet ===

Version: 4.3.1.4445

=== .NET Core ===

Runtime: Not installed
SDK: Not installed
MSBuild SDKs: /usr/lib/mono/msbuild/15.0/bin/Sdks

=== Build Information ===

Release ID: 706000711
Git revision: 634060c6c105fd3ee36678fe547680fd258cbf45
Build date: 2018-05-08 11:37:23-04

=== Operating System ===

Linux
Linux 4.15.0-20-generic #21-Ubuntu SMP Tue Apr 24 06:16:15 UTC 2018 x86_64

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-24] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: Lluis Sanchez  
 Commit: 781698a9ac9528f6532749839a7fcf42d6e99052 (xamarin/monodevelop)
